tests/unittests/TensorsTest.cpp

```diff
@@ -1238,3 +1238,31 @@ TEST(CustomAlignedTensor, getDimForPtr) {
   EXPECT_EQ(H.getDimForPtr(1, 3), 1);
   EXPECT_EQ(H.getDimForPtr(2, 3), 0);
 }
+
+TEST(Tensor, sameAlignmentDifferentValues) {
+  Type Ty1(ElemKind::Float16Ty, {2, 1}, {4, 1});
+  Type Ty2(ElemKind::Float16Ty, {2, 1}, {4, 1});
+  Tensor T1(Ty1);
+  Tensor T2(Ty2);
+  auto T1H = T1.getHandle<float16_t>();
+  auto T2H = T2.getHandle<float16_t>();
+  T1H.clear(0);
+  T2H.clear(1);
+  T1H.at({0, 0}) = T2H.at({0, 0}) = 1;
+  T1H.at({1, 0}) = T2H.at({1, 0}) = 2;
+
+  EXPECT_TRUE(T1.isEqual(T2));
```

Comment:
Can you add a case where `isEqual()` is false? Maybe just by changing on of the handles here and rechecking. (and in the differentAlignment test below).

Reply:
Done. added it to both tests.